### ISSUE TITLE
Possibility to create computational graph from Tensor gradient

### ISSUE THREAD

**Issue by Shinkaiika** [LaurentMazare/tch-rs#540]
Hello

I am trying to build computational graphs from the gradients of Tensors.

I have a working Python implementation as follows:
```python
x = torch.randn(2)
x.requires_grad = True
y = neural_net.forward(x)

# first order derivative
fd = torch.autograd.grad(y, x, create_graph=True)[0]
```

Now, `fd` has a field `grad_fn` that has the field `next_functions`. With `next_functions` I can recursively build the computational graph in Python. However, in **tch-rs** there seems to be no equivalent of `grad_fn` and `next_functions`. Am I missing something, or is there another way to build the graph?

For clarity: I need the graphs to build _symbolic expressions_ that are then used in SMT solving.

Thank you.

**Comment by LaurentMazare** [comment]
I don't think that there is an easy way to do so at the moment: I think it's likely to require exposing functions like [next_edges](https://github.com/pytorch/pytorch/blob/382ef1fda75dfce07c37a920e908ce96d99bf970/torch/csrc/autograd/function.h#L287) on the Rust side and to be able to make some introspection on these edges objects.

**Comment by Shinkaiika** [comment]
Ok, thanks for the information.

Unfortunately, I don't have time to further look into this and to implement it.
Therefore, I am closing this issue for now. I might come back to it at a later point.

Best regards.
